plugins/builds/update.js

```diff
@@ -74,14 +75,21 @@ module.exports = () => ({
                     case 'RUNNING':
                         build.startTime = (new Date()).toISOString();
                         break;
+                    // do not update meta or endTime for these cases
+                    case 'UNSTABLE':
+                    case 'BLOCKED':
+                        break;
                     default:
                         throw boom.badRequest(`Cannot update builds to ${desiredStatus}`);
                     }
 
-                    // Everyone is able to update the status
-                    build.status = desiredStatus;
-                    if (statusMessage) {
-                        build.statusMessage = statusMessage;
+                    // UNSTABLE -> SUCCESS needs to update meta and endtime.
+                    // However, the status itself cannot be updated to SUCCESS
+                    if (build.status !== 'UNSTABLE') {
+                        build.status = desiredStatus;
+                        if (statusMessage) {
```

Comment:
should `statusMessage` be restricted for UNSTABLE ?

Reply:
I don't think so. Moved it out 👍